modules/report/src/main/java/com/opengamma/strata/report/framework/format/AsciiTable.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.report.framework.format;
+
+import java.util.List;
+
+import com.google.common.base.Strings;
+
+/**
+ * An ASCII table generator.
+ * <p>
+ * Provides the ability to generate an ASCII table, typically used on the command line.
+ */
+final class AsciiTable {
+
+  //-------------------------------------------------------------------------
+  /**
+   * Restricted constructor.
+   */
+  private AsciiTable() {
+  }
+
+  /**
+   * Generates the ASCII table.
+   * 
+   * @param alignments  the table alignments
+   * @param headers  the table headers
+   * @param cells  the table cells
+   * @return the table
+   */
+  static String generate(
+      List<AsciiTableAlignment> alignments,
+      List<String> headers,
+      List<? extends List<String>> cells) {
```

Comment:
Should initially ensure that the inputs are all consistent (i.e. same number of columns throughout) to avoid an `IndexOutOfBoundsException` with no helpful message.